src/ios/BEMShakeNotifier.m

```diff
@@ -0,0 +1,431 @@
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+    
+}
+
+- (void)enableEventListener:(CDVInvokedUrlCommand*)command
+{
+    CDVPluginResult* pluginResult;
+    
+    __block NSString* eventName = command.arguments[0];
+    
+    if (isNSNull(eventName) || [eventName length] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"eventName is null or empty"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    __block NSDictionary* localNotifyConfig = command.arguments[1];
+    
+    if (isNSNull(localNotifyConfig) || [localNotifyConfig count] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"localNotifyConfig is null or empty"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    // enabling the listener means removing an entry from the muted list
+    [self removeEntry:localNotifyConfig
+             forEvent:eventName
+            withLabel:MUTED_LIST_KEY];
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+- (void)disableEventListener:(CDVInvokedUrlCommand*)command
+{
+    CDVPluginResult* pluginResult;
+    
+    __block NSString* eventName = command.arguments[0];
+    
+    if (isNSNull(eventName) || eventName == nil || [eventName length] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"eventName is null"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    __block NSDictionary* localNotifyConfig = command.arguments[1];
+    
+    if (isNSNull(localNotifyConfig) || localNotifyConfig == nil || [localNotifyConfig count] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"localNotifyConfig is null or empty"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    // disabling the listener means adding an entry to the muted list
+    [self addOrReplaceEntry:localNotifyConfig
+                   forEvent:eventName
+                  withLabel:MUTED_LIST_KEY];
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+-(void)enableSensor{
+    motionManager = [[CMMotionManager alloc] init];
+    if([motionManager isAccelerometerAvailable]){
+        if([motionManager isAccelerometerActive] == NO){
+            [LocalNotificationManager addNotification:[NSString stringWithFormat:
```

Comment:
Don't you need to actually process the accelerometer inputs before you know whether there is a shake or not?